### ISSUE TITLE
pdb history file should be different per default, not the same with main repl

### ISSUE THREAD

**Issue by skirpichev** [python/cpython#124994]
# Feature or enhancement

### Proposal:

After quitting from debug session (triggered by breakpoint()) to the main repl, I've it's history filled with a bunch of pdb-specific stuff.  Usually (always?) these lines have an invalid syntax for python.

Maybe it does make sense have access to recent repl entries after starting pdb, but hardly the main repl's history should have access to pdb history.  That should live in a separate file. 

### Has this already been discussed elsewhere?

No response given

**Comment by gaogaotiantian** [comment]
How did you trigger the pdb? From repl? So like

```
>>> breakpoint()
(pdb) quit
>>> <up-arrow>
```

Is there a case where this is actually useful? If you are in repl, you can already do plenty of interactive things and you are probably not debugging a serious script. It's hard for me to imagine a real use case.

**Comment by skirpichev** [comment]
> So like

No.  Just breakpoint() in a source code, which was run from the repl.

> you are probably not debugging a serious script.

Maybe.  But above scenario apparently does make sense for non-serious scripts.  The breakpoint() builtin does exits for reasons, eh?

**Comment by gaogaotiantian** [comment]
As far as I know, the `breakpoint()` builtin is not often used together with repl. If it's easily solvable, I'm definitely not against the proposal. If we can just ignore `pdb` history by flipping a switch, let's do it. However, if it's not a trivial task, we probably need to think about the benefit here. How are we going to separate history from `pdb`? If we do that, how about other repl-like programs? How about `input()`? We record history from `input()` now and those are rarely valid Python script. What is the cutoff here?

**Comment by skirpichev** [comment]
> If it's easily solvable

I don't know.  But IPython has no such issue: you can fall into pdb session, the quit into repl back.  No garbage in the history.  And next time, entering pdb session - I see previous pdb commands...

> If we can just ignore pdb history by flipping a switch, let's do it.

It should definitely be ignored when you quit from the pdb session back to repl.

PS: I should note, that issue is valid for old repl as well.
